Neos.ContentRepository.Core/Classes/ContentGraphFinderInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphInterface;
+use Neos\ContentRepository\Core\Projection\ProjectionStateInterface;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Exception\WorkspaceDoesNotExist;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * A finder for ContentGraphInterface bound to a Workspace
+ *
+ * @see ContentGraphInterface
+ * @internal
+ */
+interface ContentGraphFinderInterface extends ProjectionStateInterface
```

Comment:
Why do we need that interface?
You added `reset()` only to the single `ContentGraphFinder` implementation and call it from the code (e.g. in `WorkspaceCommandHandler::resetContentGraphCache()` and in Behat traits.
We should instead reference the interface (and add required methods) or remove it (which I would prefer)

Reply:
I think I originally thought to separate userland from internal implementation parts by annotating inteface or implementation respectively, BUT that is moot as the userland part is ContentRepository::getContentGraph(WorkspaceName) and that's it, I will drop it.